ai-analysis/test-08-12-25/M7_DETAILED_ISSUES_FOR_ENGINEERING.md

```diff
@@ -0,0 +1,301 @@
+
+**Severity:** HIGH  
+**Impact:** Route registration failures, developers unable to create custom pages
+
+---
+
+### Issue #2: Page Loader Repository Access Pattern (add-custom-page.md)
+
+**Location:** `guides/plugins/plugins/storefront/add-custom-page.md` (estimated line 40-50)
+
+**Documented Claim:**
+> "Page loader should not use repository directly, must use store api route instead"
+
+**Actual Code Finding:**
+- Code example shows direct repository access
+- Validated against: `platform/src/Core/Content/Cms/`
+
+**Issue:**
+Documentation violates its own stated architecture principle - example code uses repositories when it explicitly says not to
+
+**Suggested Fix:**
+1. Either update guidance to allow direct repository access (if that's current practice)
+2. Or fix code example to use Store API routes as documented
+3. Clarify when each approach is appropriate
+
+**Severity:** HIGH  
+**Impact:** Architecture confusion, inconsistent patterns across codebase
+
+---
+
+### Issue #3: Repository Search Method Signature (reading-data.md)
+
+**Location:** `guides/plugins/plugins/framework/data-handling/reading-data.md` (estimated line 30-40)
+
+**Documented Claim:**
+> "Repository search method signature: `search(Criteria, Context)`"
+
+**Actual Code Finding:**
+- Method signature has changed
+- Validated against: `platform/src/Core/Framework/DataAbstractionLayer/EntityRepository.php`
+
+**Issue:**
+Method signature in documentation doesn't match actual implementation in EntityRepository
+
+**Suggested Fix:**
+1. Check actual `EntityRepository::search()` signature in 6.7
+2. Update documentation with correct parameter types and order
+3. Add return type documentation
+4. Include working code example
+
+**Severity:** HIGH  
+**Impact:** Code won't compile, type errors, developer confusion
+
+---
+
+### Issue #4: Repository Service Naming Convention (reading-data.md)
+
+**Location:** `guides/plugins/plugins/framework/data-handling/reading-data.md` (estimated line 15-25)
+
+**Documented Claim:**
+> "Repository service name follows pattern: `entity_name.repository`"
+
+**Actual Code Finding:**
+- Service naming pattern has evolved
+- Validated against: `platform/src/Core/Framework/DependencyInjection/`
+
+**Issue:**
+Service container registration pattern doesn't match documented naming convention
+
+**Suggested Fix:**
+1. Review `services.xml` files in platform core
+2. Document actual service naming pattern
+3. Provide examples of finding service names
+4. Add console command for listing entity repositories
+
+**Severity:** MEDIUM  
+**Impact:** Service not found errors, dependency injection failures
+
+---
+
+## 🟡 MEDIUM PRIORITY ISSUES
+
+### Issue #5: App Request Headers (app-base-guide.md)
+
+**Location:** `guides/plugins/apps/app-base-guide.md` (estimated line 60-70)
+
+**Documented Claim:**
+> "Request headers include `shopware-app-signature` and `sw-version`"
+
+**Actual Code Finding:**
+- Header names don't match implementation
```

Comment:
where does it not match?
This is the code:
```
        return new Request(
            'GET',
            $uri,
            [
                'shopware-app-signature' => $signature,
                'sw-version' => $this->shopwareVersion,
            ]
        );
```
and the headers are exactly what is documented